src/text/parsers/symbol.rs

```diff
@@ -102,26 +101,7 @@ fn identifier_trailing_characters(input: &str) -> IResult<&str, &str> {
 /// Matches an operator (e.g. `++` or `@`) and returns the resulting [String]
 /// as a [TextValue::Symbol]. This symbol syntax is only recognized inside of an s-expression.
 pub(crate) fn parse_operator(input: &str) -> IResult<&str, TextValue> {
-    // This function is used by the [s_expression_value] parser in the [containers] module.
-
-    // The 'recognizer' below  is a parser responsible for identifying the &str slice at the
-    // beginning of input that represents an operator. The `map` operation that follows uses
-    // this parser's output to construct the necessary TextValue.
-    let recognizer = delimited(
-        // Other parsers don't have their own leading whitespace matcher because the overarching
-        // top_level_stream_value parser takes care of this. When matching an s-expression, this
-        // parser is given precedence over the other parsers; because of this, it must consume
-        // the whitespace on its own.
-        multispace0,
-        // `is_a` matches the longest leading string comprised of one or more of the given characters
-        is_a("!#%&*+-./;<=>?@^`|~"),
-        // The operator must be followed either by whitespace or the end of the s-expression.
-        alt((
-            peek(recognize(one_of(" \r\n\t"))),
-            peek(recognize(s_expression_end)),
-        )),
-    );
-    // The above `recognizer` outputs a &str; this operation turns that &str into a Symbol.
+    let recognizer = preceded(multispace0, is_a("!#%&*+-./;<=>?@^`|~"));
```

Comment:
🗺️ Some earlier restructuring in `src/text/parsers/containers.rs` eliminated the (wrong/unnecessary) rule that s-expression scalars have whitespace or the closing `)` after them.